### PR TITLE
Create user model

### PR DESCRIPTION
**Pull request by NOI-lgtm**
- Create User model 
- Create Active record association between User and Thought table
- Add destroy dependent so that the thought that is related to user is deleted 
- Add method to get full name in User model
- Ran migration to update schema 
Don't forget to run rails db:migrate

### CODE REVIEW

File: app/models/thought.rb [hunk 1]
```diff
@@ -1,2 +1,4 @@
 class Thought < ApplicationRecord
+  belongs_to :user
```
Comment (KellyAH) [inline app/models/thought.rb:2]: after our chat on Saturday about the DB architecture, is this still belongs_to or did we decide it's a different Rails active record association?

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  has_many :thoughts, :dependent => :destroy
+
+  def full_name
+    [first_name, last_name].join(' ')
+  end
```
Comment (KellyAH) [inline app/models/user.rb]: if full_name is going to be displayed on the FE, does it also need to be made pretty where the `first_name` and `last_name` need to be capitalized?

File: db/migrate/20201029194434_create_users.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class CreateUsers < ActiveRecord::Migration[6.0]
+  def change
+    create_table :users do |t|
+      t.string :first_name
+      t.string :last_name
+      t.string :user_name
+      t.datetime :dob
```
Comment (KellyAH) [inline db/migrate/20201029194434_create_users.rb:7]: why do we need date of birth? is this for a happy birthday message we were thinking of doing in the app?

File: db/migrate/20201029194434_create_users.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class CreateUsers < ActiveRecord::Migration[6.0]
+  def change
+    create_table :users do |t|
+      t.string :first_name
+      t.string :last_name
+      t.string :user_name
```
Comment (KellyAH) [inline db/migrate/20201029194434_create_users.rb:6]: if Users table has a `user_name` column, does that mean we need a `password` column for the future admin auth part?

File: test/models/user_test.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (KellyAH) [inline test/models/user_test.rb:6]: what is this commented out code? is it being saved for later or can it just be removed? We haven't written any tests yet right? so is uncommenting this going to break anything?

File: app/models/thought.rb (hunk 1 above)
Comment (NOI-lgtm) [reply app/models/thought.rb:2]: This is a User - Thought relationship association that we did discussed :)  

One : Many relationship  
- User has many thoughts
- Thought belongs to User 

The link to diagram https://dbdiagram.io/d/5f9b12403a78976d7b79bc81

File: app/models/user.rb (hunk 2 above)
Comment (NOI-lgtm) [reply app/models/user.rb]: Yeah, I guess we can add extra method to capitalize here :)

File: db/migrate/20201029194434_create_users.rb (hunk 3 above)
Comment (NOI-lgtm) [reply db/migrate/20201029194434_create_users.rb:7]: Yeah that's right :)

File: db/migrate/20201029194434_create_users.rb (hunk 4 above)
Comment (NOI-lgtm) [reply db/migrate/20201029194434_create_users.rb:6]: For now it is just to display as "hey user these are your thoughts" but definitely need password for authentication later.

File: test/models/user_test.rb (hunk 5 above)
Comment (NOI-lgtm) [reply test/models/user_test.rb:6]: This is auto generated test code that comes out of the box from rails mini test and I guess we can get rid of the whole mini test package and auto generated test files and replaced with Rspec :) A link about auto generated test folder in rails https://guides.rubyonrails.org/testing.html

File: app/models/user.rb (hunk 2 above)
Comment (NOI-lgtm) [reply app/models/user.rb]: May be write some test too :D

**Comment by karolina-benitez** [comment]
Just pushed changes. Getting error `SQLite3::SQLException: unknown column "user_id_id" in foreign key definition` when running migration to add foreign key to user_categories table

**Review by KellyAH — APPROVED** [review]
Reviewed this PR with Naomi and Neer. We made cleaned up some unnecessary code and added TODOs.
 LGTM

**Pull request merged**